src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq_vmr.c

```diff
@@ -403,32 +422,33 @@ static void xgq_vmr_log_dump(struct xocl_xgq_vmr *xgq, int num_recs, bool dump_t
 
 		log_idx = (log_idx + VMR_LOG_MAX_RECS - num_recs) % VMR_LOG_MAX_RECS;
 
-		if (!dump_to_debug_log)
-			XGQ_WARN(xgq, "=== start dumping vmr log ===");
-
 		for (idx = 0; idx < num_recs; idx++) {
 			xocl_memcpy_fromio(&log.log_buf, xgq->xgq_payload_base +
 				xgq->xgq_vmr_shared_mem.log_msg_buf_off +
 				sizeof(log) * log_idx,
 				sizeof(log));
 			log_idx = (log_idx + 1) % VMR_LOG_MAX_RECS;
 
-			if (dump_to_debug_log)
-				XGQ_DBG(xgq, "%s", log.log_buf);
-			else
-				XGQ_WARN(xgq, "%s", log.log_buf);
+			/* calling call back function */
+			count += dump_cb(xgq, buf ? buf + count : NULL, log.log_buf);
+			if (count > PAGE_SIZE) {
+				XGQ_WARN(xgq, "message size %ld exceeds page %ld",
+					count, PAGE_SIZE);
+				break;
+			}
 		}
-
-		if (!dump_to_debug_log)
-			XGQ_WARN(xgq, "=== end dumping vmr log ===");
 	} else {
 		XGQ_WARN(xgq, "vmr payload partition table is not available");
 	}
+
+	return min(count, PAGE_SIZE);
 }
 
 static void xgq_vmr_log_dump_all(struct xocl_xgq_vmr *xgq)
 {
-	xgq_vmr_log_dump(xgq, VMR_LOG_MAX_RECS, false);
+	XGQ_WARN(xgq, "=== start ===");
```

Comment:
could we bring the original log here?
XGQ_WARN(xgq, "=== start dumping vmr log ===");
and
XGQ_WARN(xgq, "=== end dumping vmr log ===");